### PR TITLE
upgrading Docusaurus to v3

### PR DESCRIPTION
**Pull request by Aditya-eddy**
# Pull Request Template

## Description
fixes [#1833](https://github.com/keploy/keploy/issues/1833)
We should upgrade to v3 so that we increase the browsers support. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding tests.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by Aditya-eddy** [comment]
Hey @re-Tick I  just created the pr and pushed it. Have not created an issue for it should I?

**Comment by re-Tick** [comment]
> Hey @re-Tick I just created the pr and pushed it. Have not created an issue for it should I?

It will be good to mention the issue.

File: src/theme/BlogLayout/index.js [hunk 1]
```diff
@@ -1,40 +0,0 @@
-/**
```
Comment (Hermione2408) [inline src/theme/BlogLayout/index.js:1]: Hey @Aditya-eddy what's the reason behind removing these files are they not getting used now ?

Comment (Hermione2408) [reply src/theme/BlogLayout/index.js:1]: Also can you please check this warning 

(node:187143) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
        Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
        Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
(Use `node --trace-deprecation ...` to show where the warning was created)
client (webpack 5.88.2) compiled successfully

Comment (Aditya-eddy) [reply src/theme/BlogLayout/index.js:1]: Hey @Hermione2408 about the deleted files I think they were used for blogs section. But I don't think there is any blog section in website. Let me look into the error!

**Comment by Sonichigo** [comment]
Hey @Aditya-eddy, there is an error while starting with `yarn start` && `yarn build`

image

**Comment by Aditya-eddy** [comment]
> Hey @Aditya-eddy, there is an error while starting with `yarn start` && `yarn build`
> With last commit as well the deploy check is failing :- https://github.com/keploy/docs/actions/runs/8749516911/job/24011321177?pr=374

Hey @Sonichigo actually the docusaurus uses nodejs>=18, but in the github workflow file it was downloading the 16 some version. I have corrected the version to 18.0.0 nodejs. I think the workflow went smoothly this time. But please do check and give feedback on it. Thanks!!

**Review by Sonichigo — APPROVED** [review]
image

@Aditya-eddy there are certain warning that needs to be fixed before final merge

**Comment by Aditya-eddy** [comment]
Hey @Sonichigo I have resolved all the errors only the pet-clinic error is remaining because in there we have used h1 anchors manually whiich docusaurus does not allow hence shows. Broken-Links Reference [here](https://github.com/facebook/docusaurus/issues/5036).
Please give you feedback on this. Thanks!!

**Review by Sonichigo — CHANGES_REQUESTED** [review]
In http://localhost:3000/docs/quickstart/pet-clinic/ there are two brokenAnchors warning! You can use below to update and fix those. 

- [Using Keploy's binary](#instructions-for-starting-using-binary)
- [Using Keploy's docker image](#instructions-for-starting-using-docker)

# Instructions For Starting Using Binary <a name="binary-guide"></a>

to 

## Instructions For Starting Using Binary.

and 

# Instructions For Starting Using Docker <a name="docker-guide"></a>
to
## Instructions For Starting Using Docker

**Comment by Aditya-eddy** [comment]
Hey @Sonichigo I have pushed the updated changes!

**Review by nehagup — COMMENTED** [review]
@Aditya-eddy this is great! Just that we can't make h1 tags h2. We need to find a workaround to maintain h1 tags. 
Instead of using anchors we can directly link endpoints for H1 in case anchors are not supported for H1s.

**Comment by Aditya-eddy** [comment]
Hey @nehagup i have swizzled heading component and made the h1 tags have anchors too. By default they will have id as heading but we can set them explicitly by #heading {#id}. Please give feedback on this.

**Review by nehagup — APPROVED** [review]
Thanks @Aditya-eddy Looks promising! 🚀

**Pull request merged**